plugins/xml-adapter/src/main/java/org/polypheny/db/adapter/xml/XmlToPolyConverter.java

```
@@ -0,0 +1,124 @@
+            } else if ( event == XMLStreamConstants.END_ELEMENT ) {
+                break;
+            }
+        }
+
+        if ( value.length() == 0 ) {
+            return new PolyNull();
+        }
+
+        String valueStr = value.toString();
+        return switch ( type ) {
+            case "boolean" -> new PolyBoolean( Boolean.parseBoolean( valueStr ) );
+            case "integer" -> new PolyLong( Long.parseLong( valueStr ) );
+            case "decimal" -> new PolyBigDecimal( new java.math.BigDecimal( valueStr ) );
+            case "float" -> new PolyFloat( Float.parseFloat( valueStr ) );
+            case "double" -> new PolyDouble( Double.parseDouble( valueStr ) );
+            case "date" -> {
+                LocalDate date = LocalDate.parse( valueStr, DateTimeFormatter.ISO_DATE );
+                yield new PolyDate( date.atStartOfDay().toInstant( java.time.ZoneOffset.UTC ).toEpochMilli() );
+            }
+            case "time" -> {
+                LocalTime time = LocalTime.parse( valueStr, DateTimeFormatter.ISO_TIME );
+                yield new PolyTime( (int) (time.toNanoOfDay() / 1000000) );
+            }
+            case "dateTime" -> {
+                LocalDateTime dateTime = LocalDateTime.parse( valueStr, DateTimeFormatter.ISO_DATE_TIME );
+                yield new PolyTimestamp( dateTime.toInstant( java.time.ZoneOffset.UTC ).toEpochMilli() );
+            }
+            case "base64Binary" -> {
+                byte[] binaryData = Base64.getDecoder().decode( valueStr );
+                yield new PolyBinary( binaryData, binaryData.length );
+            }
+            case "hexBinary" -> {
+                try {
+                    byte[] hexBinaryData = Hex.decodeHex( valueStr.toCharArray() );
+                    yield new PolyBinary( hexBinaryData, hexBinaryData.length );
+                } catch ( Exception e ) {
+                    yield new PolyString( valueStr ); // Fallback in case of error
```

Why is `hexBinary` not as strict as the others?

---

The decoder exception is not propagated  up to the moveNext in the enumerator where an exception is thrown.